Button Group component

AleshaOleg opened the pull request.

AleshaOleg commented:
This branch based on `button-stories`

pseudonymcp commented:
Visual QA notes: @AleshaOleg  Why does the wrapper around the group have a 16px margin? We need to be able to use the group to left- or right-align it with form fields or content without indent.
![pseudoshot-0011](https://user-images.githubusercontent.com/1522647/94779552-71f21d80-03c7-11eb-905e-15101a371d95.png)

AleshaOleg commented:
@pseudonymcp thanks! forgot to remove static margin value, now it's possible to add margin throw property and a default value is 0.

pseudonymcp reviewed (changes requested):
There are only special cases where a button and a link should be vertically stacked, but not multiple button components themselves. We will handle this case on another side, but not inside the component, so let's remove the vertical aligment for the group.  

- Remove the `vertical` and `verticalAlign` props, and also remove those aligments in the Aligment story.
- In the Aligment story there is a 16px margin around the group. Where does this come from? There should be no margin.

tobiaslins reviewed (changes requested):
* Why does the group component have a `width` prop? This should always be auto and should be removed entirely!
* There is no `size` prop present as defined in https://jira.ubergrape.com/browse/GRAPE-18602
should the `ButtonGroup` have a size attribute that's passed to all buttons automatically? @pseudonymcp

pseudonymcp commented:
Yes, size should be for the whole group. Either all buttons regular (default), or all buttons small, can't be mixed. So if there are mixed sizes defined for the buttons withon a group, basically the button group size should override any differently defined size within one of the buttons. Can we do this? @tobiaslins @AleshaOleg 

> * There is no `size` prop present as defined in https://jira.ubergrape.com/browse/GRAPE-18602
>   should the `ButtonGroup` have a size attribute that's passed to all buttons automatically? @pseudonymcp

AleshaOleg commented:
Thanks, just removed it. Added it before to controlling width for groups which aligned vertically.

> Why does the group component have a width prop? This should always be auto and should be removed entirely!

tobiaslins commented:
@AleshaOleg `size` property is still missing. 
Please export the size type from the button component and use it within button-group!
Iterate over children & append the size attribute to the buttons!

AleshaOleg commented:
As we are using buttons as children for this component, so each button renders separately and have their own properties. I thought as we discussed, such rules will be defined in the usage guide. Like use buttons of the same size, use only one primary button and basic, don't use danger button and etc.. But in general yes, we can add `size` property for this component.

> Yes, size should be for the whole group. Either all buttons regular (default), or all buttons small, can't be mixed. So if there are mixed sizes defined for the buttons withon a group, basically the button group size should override any differently defined size within one of the buttons. Can we do this? @tobiaslins @AleshaOleg
> 
> > * There is no `size` prop present as defined in https://jira.ubergrape.com/browse/GRAPE-18602
> >   should the `ButtonGroup` have a size attribute that's passed to all buttons automatically? @pseudonymcp

AleshaOleg commented:
> @AleshaOleg `size` property is still missing.
> Please export the size type from the button component and use it within button-group!
> Iterate over children & append the size attribute to the buttons!

But what if the developer will set two buttons of different sizes in the same group?

tobiaslins commented:
> > @AleshaOleg `size` property is still missing.
> > Please export the size type from the button component and use it within button-group!
> > Iterate over children & append the size attribute to the buttons!
> 
> But what if the developer will set two buttons of different sizes in the same group?

should be overwritten by button group property as it's from the guidelines that different shouldn't be possible.

But it's okay for me to keep it in Buttons only and just have it as guideline. @pseudonymcp

pseudonymcp commented:
In the usage guidelines we'll define that there should only be one primary button, and the order of buttons (primary, basic, ...). Mixing sizes is tricky, but ok, let's keep it open and not put a `size` prop in the button group.

tobiaslins commented on web/src/stories/button-group/base.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+  </ButtonGroup>
+)
+
+Base.parameters = {
+  design: {
+    type: 'figma',
+    url:
+      'https://www.figma.com/file/Mc03kFcaFlcMOA7afjaZmF/Grape-DS-Components-Web?node-id=385%3A6',
+  },
+}
```
this can be moved into the default export (until we point to different figma files/nodes) seems to be the same for all!

pseudonymcp replied:
@tobiaslins The Figma url is different for every component.

tobiaslins commented:
pseudonym <notifications@github.com> schrieb am So. 11. Okt. 2020 um 21:36:

> *@pseudonymcp* commented on this pull request.
> ------------------------------
>
> In web/src/stories/button-group/base.stories.tsx
> <https://github.com/ubergrape/grape-ds/pull/23#discussion_r502956597>:
>
> > +Base.parameters = {
> +  design: {
> +    type: 'figma',
> +    url:
> +      'https://www.figma.com/file/Mc03kFcaFlcMOA7afjaZmF/Grape-DS-Components-Web?node-id=385%3A6',
> +  },
> +}
>
> @tobiaslins <https://github.com/tobiaslins> The Figma url is different
> for every component.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ubergrape/grape-ds/pull/23#discussion_r502956597>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAWXIPAWFYJHPKXKFPGQ6TDSKICKHANCNFSM4R7VSVRQ>
> .
>
